### ISSUE TITLE
Directory Cloning Post Build

### ISSUE THREAD

**Issue by ParchmentWolf** [Damian666/Orion-Plus-Game-Engine#42]
Dont do that xD as a distributor who isnt prepackaging content for expected use in the data files, just add the tag to copy if newer, cloning every time post build no matter what has 2 really bad downsides, 1) Debug and Compile time go way the hell up, 2) Harddrives, even Solid State Drives only last so many rewrites on their sector blocks... Constant rewrite makes those sectors go dead way faster meening you are rotting anyones harddrives really fast who uses it. Like legit you start an app and have a couple min test tops, have to constantly restart it or recompile it for hours, days? Just imagine how much damage is caused from those mass rewrites...

Put simply as i started with this, Dont. Do. That.

**Comment by SpiceyWolf** [comment]
I can imagine a reason you (or anyone helping you who may have been responsible) didnt like having to navigate menus over and over per file u wanna add... that can be painfully slow for even just 1 file. Visual Studio has always come with a option u can click on the solution explorer to show files that exist in those directories but arent necessarily "included" in the project.

Click the project name, and as you do what what lights up at the top of the solution explorer window, one of the lit up options will say "Show All Files" it may or may not look like 3 papers depending on the VS you are using, click that and for that project alone itll show the hidden files. Right click on them(You can even use Shift/Ctrl + Click to select multiple first) and select "Include in project"

Now its part of the project and you can select to copy if newer simply as that.

**Comment by SpiceyWolf** [comment]
Alternatively, Pay attention to code you are adding later on, If it uses the keyword "And" or "Or" make sure it is infact trying to perform BitWise checks(Like the directional block system does). If not make sure to replace "And" with "AndAlso" and "Or" with "OrElse". If you dont, the compiler has to correct it at runtime, that comes with a delay before it even realizes the problem, and before it can figure out how to correct it, then move on. And this isnt a permanent correction, this is a at runtime correction. This meens if u call a function that has that error or use it in a loop, everytime the function or loop is called it has to redo the correction which is already super slow as is to do, but doing it rapidly just kills all performance. 

Our friend Krilliac found one example in the Tile checks for if its a key or door, it was doing TileX by TileY in a loop checking with a bitwise operator, on a map that is 30x30, thats 900 times a loop (that is called every 5 seconds) that it has to perform that code, wait for a crash to happen durring polling, then it has to diagnose the crash, detect solution options, apply them, then rerun the same freakin line it just crashed on.

It had something like a 12% of performance cost in runtime was eaten up by that alone. -> Something that otherwise should REALLY cost virtually no performance like less than 1% definitely.

**Comment by SpiceyWolf** [comment]
Im going to go through and correct these where i can find them, but please be mindful in the future :) Grow as a programmer and take the extra mile to do small things like these, dont just ignore them cause "it works" like eclipse has through its entire existence.

**Comment by SpiceyWolf** [comment]
I looked through and only seem to be able to find the dirblock actually needing the bitwise operator, SO i mass changed everything and got about 1k times the wrong thing was used xD
also 658 incorrect ors :D

**Comment by Raven-Quest** [comment]
and i dunno why i did that, was easier i guess? if it was me in the first place.

those checks are just what is was to begin with, I did not know better tbh...

so what you are saying is only bitwise, aka 1 or 0 should be handled that way?

I'm trying to let go of the  eclipse style coding, but as you can see, its proving hard for me >.<

**Comment by SpiceyWolf** [comment]
I didnt assume u did know better, but i care enough to see you be a better programmer i informed you and took care of them while your source code is in my care for now :)

**Comment by SpiceyWolf** [comment]
yeah only use the bitwise with And and Or operators.

**Comment by SpiceyWolf** [comment]
if its simple conditions, must be AndAlso and OrElse otherwise the runtime has to convert it and that is really bad.

**Comment by SpiceyWolf** [comment]
Hey also im sorry you took offense to what i said about you on jc's website, it wasnt meant as an insult to you, it was moreso me trying to explain that hey, you and jc arent the most reliable to look to for information on coding cause you guys dont have all the answers, you certainly dont know a majority of the low level or performance on steroids stuff i do... thats not cause you are stupid you just dont have that knowlege, JC doesnt have it either but he likes to claim shit when he doesnt know what hes talking about... toward him yeah it was intended to be offensive cause hes a lying asshole... u dont claim to have all those answers so i dont wish to throw you in a box like that of smug asshole or something xD

**Comment by Raven-Quest** [comment]
yea, I'm a noob, and I know it :P

**Comment by SpiceyWolf** [comment]
Its fine though man, the progress you made on this engine even with help is still impressive non the less :D
